CHANGELOG.rst

```diff
@@ -5,49 +5,48 @@ Changelog
 TBR
 ---
 
-* Scrapy uses the spider's ``parse()`` method when the :class:`scrapy.http.Request`
-  callback is set to ``None``.
+* Improved ``scrapy-poet`` behavior when ``spider.parse()`` method
+  arguments are annotated. Previously, they were causing unnecessary work
+  in unexpected places like
+  :class:`scrapy.downloadermiddlewares.robotstxt.RobotsTxtMiddleware`,
+  :class:`scrapy.pipelines.images.ImagesPipeline` or
+  :class:`scrapy.pipelines.files.FilesPipeline`. It is also a reason
+  ``HttpClient`` might not be working in page objects.
+  Now these cases are detected, and a warning is issued.
+
+  As of Scrapy 2.7, it is not possible to fix the issue completely
+  in scrapy-poet. Fixing it would require Scrapy changes; some 3rd party
+  libraries may also need to be updated.
+
+  .. note::
+
+      The root of the issue is that when request.callback is None, ``parse``
+      callback is assumed normally. But sometimes callback=None is used
+      when :class:`scrapy.http.Request` is added to the Scrapy's downloader
+      directly, in which case no callback is used. Middlewares, including
+      scrapy-poet's, can't distinguish between these two cases, which causes
+      all kinds of issues.
+
+  We recommend all ``scrapy-poet`` users to modify their code to
```

Comment:
```suggestion
  We recommend all **scrapy-poet** users to modify their code to
```

I think we should probably format it like this so it doesn't look like the inline code we have.